dotcom-rendering/src/lib/getFrontsBannerAdPositions.test.ts

```diff
@@ -0,0 +1,237 @@
+			collectionType: 'fixed/small/slow-IV',
+			containerPalette: undefined,
+			displayName: 'The rural network',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'dynamic/fast',
+			containerPalette: undefined,
+			displayName: 'Around the world',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/thrasher',
+			containerPalette: undefined,
+			displayName: 'Contact the Guardian',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/video',
+			containerPalette: undefined,
+			displayName: 'Videos',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/medium/slow-VI',
+			containerPalette: undefined,
+			displayName: 'Multimedia',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/small/slow-IV',
+			containerPalette: 'Branded',
+			displayName: 'Guardian Labs',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/medium/slow-XII-mpu',
+			containerPalette: undefined,
+			displayName: 'Explore',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'fixed/small/slow-I',
+			containerPalette: undefined,
+			displayName: 'The big picture',
+			grouped: defaultGrouped,
+		},
+		{
+			collectionType: 'news/most-popular',
+			containerPalette: undefined,
+			displayName: 'Most viewed',
+			grouped: defaultGrouped,
+		},
+	];
+
+	it('calculates ad positions correctly', () => {
+		const adPositions = getFrontsBannerAdPositions(testCollections, 'uk');
+
+		expect(adPositions).toEqual([3, 6, 9, 12, 16, 19]);
+	});
+
+	it('does NOT insert ads above or below branded content', () => {
+		const brandedTestCollections: Pick<
+			DCRCollectionType,
+			'collectionType' | 'containerPalette' | 'displayName' | 'grouped'
+		>[] = [
+			{
+				collectionType: 'dynamic/fast',
+				containerPalette: undefined,
+				displayName: 'Headlines',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'fixed/small/slow-IV',
+				containerPalette: 'Branded',
+				displayName: 'Guardian Labs',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'dynamic/fast',
+				containerPalette: undefined,
+				displayName: 'Headlines',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'fixed/small/slow-IV',
+				containerPalette: 'Branded',
+				displayName: 'Guardian Labs',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'dynamic/fast',
+				containerPalette: undefined,
+				displayName: 'Headlines',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'fixed/small/slow-IV',
+				containerPalette: 'Branded',
+				displayName: 'Guardian Labs',
+				grouped: defaultGrouped,
+			},
+			{
+				collectionType: 'dynamic/fast',
+				containerPalette: undefined,
+				displayName: 'Headlines',
+				grouped: defaultGrouped,
+			},
+		];
+
+		const adPositions = getFrontsBannerAdPositions(
+			brandedTestCollections,
+			'uk',
+		);
+
+		expect(adPositions).toEqual([]);
+	});
+});
+
+describe.only('Tagged fronts fronts-banner ad slots', () => {
```

Comment:
Is this `only` a leftover?

Reply:
Yes I would have been very sad had this made it through to main! Thanks for the catch.